src/main/java/edu/oregonstate/cs361/battleship/HardBattleshipModel.java

```diff
@@ -79,23 +87,93 @@ else if (orient == 1) {                       // if vert, row restricted by leng
             }
         }
 
-        /* System.out.println("Ship locations after random placement: ");
+        /*System.out.println("Ship locations after random placement: ");
         for (Ship ship: comp_ships) {
             System.out.println(ship.getName() + ": starts " + ship.start.getAcross() + ", " + ship.start.getDown() + "; ends " + ship.end.getAcross() + ", " + ship.end.getDown());
         }
         System.out.println(); */
+
     }
 
     @Override
     public void shootAtPlayer() {
-        // TODO: smart shooting. Pew pew
-        int max = 10;
-        int min = 1;
-        Random random = new Random();
-        int randRow = random.nextInt(max - min + 1) + min;
-        int randCol = random.nextInt(max - min + 1) + min;
 
-        Coordinate coor = new Coordinate(randRow,randCol);
-        playerShot(coor);
+        //if the AI has not hit a ship fire randomly until it has
+        if (randToggle == true) {
+            int max = 10;
+            int min = 1;
+            Random random = new Random();
+            int randRow = random.nextInt(max - min + 1) + min;
+            int randCol = random.nextInt(max - min + 1) + min;
+
+            Coordinate coor = new Coordinate(randRow, randCol);
+            playerShot(coor);
+
+            //if one of the following ships was hit begin hitting the rest of the ship
+            if (aircraftCarrier.covers(coor)) {
+                setShipInfo(getShip("aircraftcarrier"), coor);
+            }
+            if (battleship.covers(coor)) {
+                setShipInfo(getShip("battleship"), coor);
+            }
+            if (submarine.covers(coor)) {
+                setShipInfo(getShip("submarine"), coor);
+            }
+            return;
+        }
+
+        //if the AI hit a ship continue hitting it
+        if (randToggle == false) {
+
+            if (horizontal == false){
+                if (nextHit.equals(firstHit)){
+                    nextHit.setAcross(nextHit.getAcross() + 1);
+                }
+            }
+
+            if (horizontal == true) {
+                if (nextHit.equals(firstHit)) {
+                    nextHit.setDown(nextHit.getDown() + 1);
+                }
+            }
+
+            Coordinate coor = new Coordinate(nextHit.getAcross(), nextHit.getDown());
+            playerShot(coor);
+            numHit++;
+
+
+            //if the AI has hit all the places on the ship reset the ship information so it will randomly fire again
+            if (numHit == length){
+                resetShipInfo();
+                return;
+            }
+
+            if (horizontal == false){
+                nextHit.setAcross(nextHit.getAcross() + 1);
+            }
+
+            if (horizontal == true){
+                nextHit.setDown(nextHit.getDown() + 1);
+            }
+
+            return;
+        }
+    }
+
+    void setShipInfo(Ship currentShip, Coordinate coor) {
+        length = currentShip.length;
```

Comment:
Not a necessity, but you can use "this." to make sure the proper variable is getting adjusted in the code to make sure things don't go wrong.